ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/contentassist/BuildPropertiesContentAssistProcessor.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.pde.internal.ui.editor.contentassist;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import org.eclipse.jface.text.*;
+import org.eclipse.jface.text.contentassist.*;
+import org.eclipse.pde.internal.build.IBuildPropertiesConstants;
+import org.eclipse.pde.internal.ui.editor.PDESourcePage;
+
+public class BuildPropertiesContentAssistProcessor extends TypePackageCompletionProcessor
+		implements ICompletionListener {
```

Comment:
Is there a reason why this class implements `ICompletionListener`, but leaves all of its methods unimplemented?
To me it looks like this can be removed again, together with the empty method. If that is wrong the methods should at least have the TODOs removed and there should be a comment explaining why they are empty.

Reply:
@HannesWell - thanks. `ContentAssistant.addCompletionListener(ICompletionListener listener)`  was a necessity for the auto-completion key stroke call back to work, but looks like that is not the case, this is for additional listeners when auto-completion is being carried out. So I removed it, as I don't see an associated action on this, at the moment.